Userland/Libraries/LibGfx/ImageFormats/WebPLoader.cpp

```diff
@@ -297,6 +298,352 @@ static ErrorOr<VP8LHeader> decode_webp_chunk_VP8L_header(WebPLoadingContext& con
+            return context.error("WebPImageDecoderPlugin: VP8X and VP8L chunks store different widths");
+        if (vp8l_header.height != context.vp8x_header.height)
+            return context.error("WebPImageDecoderPlugin: VP8X and VP8L chunks store different heights");
+        if (vp8l_header.is_alpha_used != context.vp8x_header.has_alpha)
+            return context.error("WebPImageDecoderPlugin: VP8X and VP8L chunks store different alpha");
+    }
+
+    FixedMemoryStream memory_stream { vp8l_chunk.data.slice(5) };
+    LittleEndianInputBitStream bit_stream { MaybeOwned<Stream>(memory_stream) };
+
+    // image-stream = optional-transform spatially-coded-image
+
+    // https://developers.google.com/speed/webp/docs/webp_lossless_bitstream_specification#4_transformations
+    // https://developers.google.com/speed/webp/docs/webp_lossless_bitstream_specification#72_structure_of_transforms
+
+    // optional-transform   =  (%b1 transform optional-transform) / %b0
+    if (TRY(bit_stream.read_bits(1)))
+        return context.error("WebPImageDecoderPlugin: VP8L transform handling not yet implemented");
+
+    // https://developers.google.com/speed/webp/docs/webp_lossless_bitstream_specification#623_decoding_entropy-coded_image_data
+    // https://developers.google.com/speed/webp/docs/webp_lossless_bitstream_specification#523_color_cache_coding
+    // spatially-coded-image =  color-cache-info meta-prefix data
+
+    // color-cache-info      =  %b0
+    // color-cache-info      =/ (%b1 4BIT) ; 1 followed by color cache size
+    bool has_color_cache_info = TRY(bit_stream.read_bits(1));
+    u16 color_cache_size = 0;
+    u8 color_cache_code_bits;
+    dbgln_if(WEBP_DEBUG, "has_color_cache_info {}", has_color_cache_info);
+    Vector<ARGB32, 32> color_cache;
+    if (has_color_cache_info) {
+        color_cache_code_bits = TRY(bit_stream.read_bits(4));
+
+        // "The range of allowed values for color_cache_code_bits is [1..11]. Compliant decoders must indicate a corrupted bitstream for other values."
+        if (color_cache_code_bits < 1 || color_cache_code_bits > 11)
+            return context.error("WebPImageDecoderPlugin: VP8L invalid color_cache_code_bits");
+
+        color_cache_size = 1 << color_cache_code_bits;
+        dbgln_if(WEBP_DEBUG, "color_cache_size {}", color_cache_size);
+
+        TRY(color_cache.try_resize(color_cache_size));
+    }
+
+    // https://developers.google.com/speed/webp/docs/webp_lossless_bitstream_specification#622_decoding_of_meta_prefix_codes
+    // "Meta prefix codes may be used only when the image is being used in the role of an ARGB image."
+    // meta-prefix           =  %b0 / (%b1 entropy-image)
+    bool has_meta_prefix = TRY(bit_stream.read_bits(1));
+    dbgln_if(WEBP_DEBUG, "has_meta_prefix {}", has_meta_prefix);
+    if (has_meta_prefix) {
+        return context.error("WebPImageDecoderPlugin: VP8L meta_prefix not yet implemented");
+    }
```

Comment:
Brackets not necessary for single line if block (and makes it more consistent with the rest of your code ;-))